### ISSUE TITLE
Please support SafeInCloud import

### ISSUE THREAD

**Issue by szaimen** [marius-wieschollek/passwords#173]
**Current Status**
I am not able to import my exported SafeInCloud csv-file to nextcloud-passwords.

**Feature Description**
I should be able to choose "SafeInCloud CSV" from the format-list for importing.

**Additional context**
The official website is: https://safe-in-cloud.com/en/
The Desktop-applications are for free and you should be able to test the CSV-export-function with it to setup a preconfigured import format.
I already tried to import via the "Other CSV" format option. But it doesn't work.

**Comment by marius-wieschollek** [comment]
I will add the profile in the august version. 

In the mean time you can just choose "Other CSV" in the import options. In step 2 just choose the file and leave the rest to default. In step 3 choose the following "CSV Field Mapping": "Label", "Username", "Password", "Url", "Notes". Leave the rest to default and press "Import"

**Comment by szaimen** [comment]
Thank your for your answer and hint. 
I followed your instructions but an error occurs when I start the import process:

`Cannot read property 'length' of undefined`

Also choosing another row in the dropdown menu in step 3: e.g. "show row 2" doesn't do anything.

Probably part of that is the error message when choosing the csv-file in step 2: 
`The file has the type "application/vnd.ms-excel" but "text/csv" is expected. You might have chosen the wrong file or importer.`

**Comment by marius-wieschollek** [comment]
Can you open the develoepr tools (press F12) and check if there are more details for the `Cannot read property 'length' of undefined` error?

The second error is normal since some OS sometimes report a wrong file type. (Except if your file really isn't a CSV. To verify that, just open it with the editor and check if it looks like a CSV)

**Comment by szaimen** [comment]
> Can you open the develoepr tools (press F12) and check if there are more details `for the Cannot read property 'length' of undefined` error?

Yes, here they are: 
![Anmerkung 2019-07-15 230809](https://user-images.githubusercontent.com/42591237/61249406-734d1100-a755-11e9-890d-c2c9b4445cff.jpg)

**Comment by marius-wieschollek** [comment]
Can you please open the file in a text editor (Not Word, something like Notepad) and check its contents?

It should look something like this:
```
"Title","Login","Password","URL","Notes"
"Facebook (Sample)","john555@gmail.com","early91*Fail*","https://www.facebook.com",""
"Google (Sample)","john555@gmail.com","plain79{Area{","https://www.google.com",""
"Laptop (Sample)","john555","Save63\apple\","",""
"test","login","password","https://example.com","Some notes for this password."
"Twitter (Sample)","john555@gmail.com","Best23^Get^","https://twitter.com",""
```

**Comment by szaimen** [comment]
So after a few hours of testing, I found out, that line-breaks are allowed in the notes-field in SafeInCloud. But they are also exported to the csv-file so that the import doesnt work until you have manually corrected the csv-file.
So the bad lines looked like this:
```
"Label","Username","Password","URL","Notes
Notes-Continuation in the next line (because of the line-break)"
```
When I corrected the bad lines e.g. to:
```
"Label","Username","Password","URL","Notes; Notes-Continuation in the same line"
```
Everything worked like a charm.

BTW: I found out, that importing just one password (header-line with one password line) doesnt work. (mount error)

**Comment by marius-wieschollek** [comment]
Thanks for your feedback.

Usually CSV fields can have more than one line if they are correctly encapsulated.
I took a closer look at the SafeInCloud export file and noticed that they usually use Windows formatted line breaks (CRLF). But sometimes they use Linux formatted line breaks (LF) for notes.
This confuses the autodetection of linebreaks and the csv parser.

I have added an option to specify the line break in the custom import and updated the SafeInCloud profile to enforce CRLF automatically.

I have also made sure that errors from the parser are no longer ignored and displayed correctly.
